File: atcoder/hhkb2020C_neq_min.rs
// unihernandez22
// https://atcoder.jp/contests/hhkb2020/tasks/hhkb2020_c
// data structures

use std::io::stdin;
use std::collections::BTreeSet;

fn main() {
    let mut line = String::new();
    stdin().read_line(&mut line).unwrap();
    let n: i64 = line.trim().parse().unwrap();

    line = String::new();
    stdin().read_line(&mut line).unwrap();
    let p: Vec<i64> = line
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();

    let mut bts = BTreeSet::new();
    
    for i in 0 .. n+1 {
        bts.insert(i);
    }

    for i in &p {
        bts.remove(i);
        println!("{}", bts.iter().next().unwrap());
    }
}

File: atcoder/hhkb2020C_neq_min.cpp
// vicfred
// https://atcoder.jp/contests/hhkb2020/tasks/hhkb2020_c
// set, ordered set, balanced tree, data structure
#include <iostream>
#include <set>

using namespace std;

int main() {
    const int maxn = 200000;

    set<int> numbers;
    for(int i = 0; i <= maxn; i++) {
        numbers.insert(i);
    }

    int n;
    cin >> n;

    int p;
    for(int i = 0; i < n; i++) {
        cin >> p;
        auto it = numbers.find(p);
        if(it != numbers.end())
            numbers.erase(it);
        cout << *numbers.begin() << endl;
    }
}

